test/smoke/smoke_test.go

```diff
@@ -203,4 +224,81 @@ var _ = Describe("Smoke test", func() {
 			}, ConsistentDuration, ConsistentPollInterval).Should(Equal(splcommon.PhaseReady))
 		})
 	})
+
+	Context("Multisite cluster deployment (M1 - multisite indexer cluster)", func() {
+		It("can deploy multistie indexers cluster", func() {
+
+			siteCount := 3
+			err := deployment.DeployMultisiteCluster(deployment.GetName(), 1, siteCount)
+			Expect(err).To(Succeed(), "Unable to deploy cluster")
+
+			// Ensure the part containing the cluster-master go to Ready phase
+			idc := &enterprisev1.IndexerCluster{}
+			Eventually(func() splcommon.Phase {
+				err := deployment.GetInstance(deployment.GetName(), idc)
+				if err != nil {
+					return splcommon.PhaseError
+				}
+				testenvInstance.Log.Info("Waiting for cluster-master instance status to be ready", "instance", idc.ObjectMeta.Name, "Phase", idc.Status.Phase)
+				dumpGetPods(testenvInstance.GetName())
+				// Test ClusterMasterPhase, as Phase is always Ready with replicas = 0 (no indexers)
+				return idc.Status.ClusterMasterPhase
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// In a steady state, we should stay in Ready and not flip-flop around
+			Consistently(func() splcommon.Phase {
+				_ = deployment.GetInstance(deployment.GetName(), idc)
+				return idc.Status.ClusterMasterPhase
+			}, ConsistentDuration, ConsistentPollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// Ensure the indexers of all sites go to Ready phase
+			siteIndexerMap := map[string][]string{}
+			for site := 1; site <= siteCount; site++ {
+				siteName := fmt.Sprintf("site%d", site)
+				instanceName := fmt.Sprintf("%s-%s", deployment.GetName(), siteName)
+				siteIndexerMap[siteName] = []string{fmt.Sprintf("splunk-%s-indexer-0", instanceName)}
```

Comment:
In the future, if we are considering more than one indexer/site, I guess, this needs to be reworked.... Wondering if its worth to have at least the SF/RF number of indexers?

Reply:
@sgontla I will pull this code out to test util package in future and will refactor this code.